apps/common/lib/lexical/source_file/store.ex

```diff
@@ -1,29 +1,40 @@
 defmodule Lexical.SourceFile.Store do
+  @moduledoc """
+  A backing store for source file documents
+
+  This implementation stores documents in ETS, and partitions read and write operations. Read operations are served
+  immediately by querying the ETS table, while writes go through a GenServer process (which is the owner of the ETS table).
+  """
   defmodule State do
     alias Lexical.SourceFile
     alias Lexical.SourceFile.Store
     require Logger
 
-    defstruct source_files: %{}, temp_files: %{}, temporary_open_refs: %{}
+    defstruct temporary_open_refs: %{}
+
+    @table_name SourceFile.Store
+
     @type t :: %__MODULE__{}
     def new do
+      :ets.new(@table_name, [:named_table, :set, :protected, write_concurrency: true])
```

Comment:
Should we also set `read_concurrency` to true? It's false by default, the docs seem to suggest that's a good thing if one expects a fair amount of reads? Disclaimer: I'm no ETS expert ;)

Reply:
Definitely. This was a mistake